### ISSUE TITLE
3.00 compilation warnings

### ISSUE THREAD

**Issue by Berbe** [transmission/transmission#1414]
# General information
OS: Linux 4.19.0-10-amd64 #1 SMP Debian 4.19.132-1 (2020-07-24) x86_64 GNU/Linux
Version: 3.00 (https://github.com/transmission/transmission/commit/bb6b5a062ee594dfd4b7a12a6b6e860c43849bfd)

# Expected
No compilation warnings

# Symptoms
Compilation warnings :
1.
```
  CCLD     libevent.la
ar: `u' modifier ignored since `D' is the default (see `U')
```
2.
```
  CCLD     libevent_openssl.la
ar: `u' modifier ignored since `D' is the default (see `U')
```
3.
```
  CCLD     libevent_core.la
ar: `u' modifier ignored since `D' is the default (see `U')
```
4.
```
  CC       test/regress-regress_buffer.o
In file included from test/regress.h:35,
                 from test/regress_buffer.c:64:
test/regress_buffer.c: In function ‘test_evbuffer_pullup_with_empty’:
test/tinytest_macros.h:116:15: warning: cast from function call of type ‘unsigned char *’ to non-matching type ‘long int’ [-Wbad-function-cast]
  type val1_ = (type)(a);      \
               ^
test/tinytest_macros.h:144:2: note: in expansion of macro ‘tt_assert_test_fmt_type’
  tt_assert_test_fmt_type(a,b,str_test,type,test,type,fmt, \
  ^~~~~~~~~~~~~~~~~~~~~~~
test/tinytest_macros.h:158:2: note: in expansion of macro ‘tt_assert_test_type’
  tt_assert_test_type(a,b,#a" "#op" "#b,long,(val1_ op val2_), \
  ^~~~~~~~~~~~~~~~~~~
test/regress_buffer.c:444:2: note: in expansion of macro ‘tt_int_op’
  tt_int_op(evbuffer_pullup(buf, -1), ==, NULL);
  ^~~~~~~~~
```
5.
```
  CCLD     libevent_extra.la
ar: `u' modifier ignored since `D' is the default (see `U')
```
6.
```
  CCLD     libevent_pthreads.la
ar: `u' modifier ignored since `D' is the default (see `U')
```
7.
```
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_CXX_FLAGS
```
8.
```
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_C_FLAGS
```
9.
```
[ 33%] Building CXX object CMakeFiles/utp.dir/utp.cpp.o
In file included from /transmission/build/third-party/utp/src/utp/utp.cpp:78:
/transmission/build/third-party/utp/src/utp/utp_config.h:11:2: warning: #warning implement this in libtransmission [-Wcpp]
 #warning implement this in libtransmission
  ^~~~~~~
```
10.
```
[ 66%] Building C object CMakeFiles/natpmp.dir/natpmp.c.o
In file included from /usr/include/x86_64-linux-gnu/bits/libc-header-start.h:33,
                 from /usr/include/string.h:26,
                 from /transmission/build/third-party/natpmp/src/natpmp/natpmp.c:32:
/usr/include/features.h:184:3: warning: #warning "_BSD_SOURCE and _SVID_SOURCE are deprecated, use _DEFAULT_SOURCE" [-Wcpp]
 # warning "_BSD_SOURCE and _SVID_SOURCE are deprecated, use _DEFAULT_SOURCE"
   ^~~~~~~
```
11.
```
[ 73%] Building C object libtransmission/CMakeFiles/transmission-test-utils.dir/utils-test.c.o
/transmission/libtransmission/utils-test.c: In function ‘test_strjoin’:
/transmission/libtransmission/utils-test.c:108:11: warning: ‘in5’ may be used uninitialized in this function [-Wmaybe-uninitialized]
     out = tr_strjoin(in5, 0, "a");
           ^~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by Berbe** [comment]
Invalid issue points:
1-3 & 5-6: [`libtool` problem in Debian](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=864018). Fixed in `2.4.6-12`: pull version from releases >= `buster`, cf. https://tracker.debian.org/pkg/libtool
4, 11: Not reproduced building from `transmission/libevent` master

Still current:
7-8: Still there
9-10: Willingful warnings, probably already monitored/known/being worked on

**Comment by ckerr** [comment]
Closing this issue as obsolete post-C++ rewrite.

If there are further compilation warnings in `main` please open a new issue.
